Fix TBB builds

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27925 Add CI builds
* #27947 Relax set_num_threads restriction in parallel native case
* **#27937 Fix TBB builds**

Summary:
Fix TBB buidls

Test Plan:
ATEN_THREADING=TBB USE_TBB=1 python setup.py develop --cmake

Differential Revision: [D17916565](https://our.internmc.facebook.com/intern/diff/D17916565)

Commit:
Fix TBB builds

Summary:
Fix TBB buidls

Test Plan:
ATEN_THREADING=TBB USE_TBB=1 python setup.py develop --cmake

[ghstack-poisoned]

ezyang commented:
cc @dylanbespalko (we should probably keep your template argument renames anyway though, it makes our library more portable)

Commit:
Update on "Fix TBB builds"


Summary:
Fix TBB buidls

Test Plan:
ATEN_THREADING=TBB USE_TBB=1 python setup.py develop --cmake

Differential Revision: [D17916565](https://our.internmc.facebook.com/intern/diff/D17916565)

[ghstack-poisoned]
